### PR TITLE
Wireguard user space implementation

### PR DESCRIPTION
**Pull request by soffokl**

### CODE REVIEW

File: services/wireguard/endpoint/userspace/client.go [hunk 1]
```diff
@@ -18,27 +18,115 @@
...
+	key, err := base64stringTo32ByteArray(config.PrivateKey())
+	if err != nil {
+		return err
+	}
+
+	if err := c.devAPI.SetPrivateKey(device.NoisePrivateKey(key)); err != nil {
+		return err
+	}
+
+	c.devAPI.Boot()
```
Comment (zolia) [inline services/wireguard/endpoint/userspace/client.go]: What this Boot performs? Can't it fail?

**Comment by zolia** [comment]
Looks like we are using original name "git.zx2c4.com/wireguard-go/", but substituting it with our fork. IMHO this is bogus, since we depend on package manager. Those who will fetch our sources and try to use other packaging might go into trouble.

I think we should use our package naming (forked) since it is very confusing and in general our source does not comply contract implemented by "git.zx2c4.com/wireguard-go/".

File: cmd/commands/service/serviceTypes.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// +build !linux linux,android
+// +build windows linux,android
```
[the remark this answers is not in the document]
Comment (soffokl) [reply cmd/commands/service/serviceTypes.go]: This is for starting services only, wireguard for Linux will work in kernel space, but it's not allowed for windows or android.

File: services/wireguard/endpoint/userspace/client.go (hunk 1 above)
Comment (soffokl) [reply services/wireguard/endpoint/userspace/client.go]: It brings the device up: https://github.com/mysteriumnetwork/wireguard-go/blob/master/device/exported_device.go#L101
In the current implementation, it returns nothing.

File: services/wireguard/endpoint/userspace/client.go [hunk 3]
```diff
@@ -18,27 +18,115 @@
...
+	if err != nil {
+		return err
+	}
+
+	extPeer := device.ExternalPeer{
+		PublicKey:  device.NoisePublicKey(key),
+		AllowedIPs: []string{"0.0.0.0/0"},
+	}
+
+	if ep := peer.Endpoint(); ep != nil {
```
[the remark this answers is not in the document]
Comment (soffokl) [reply services/wireguard/endpoint/userspace/client.go]: `nil` endpoint is a valid value, the peer can be added without an endpoint, and this endpoint will be discovered on the first connection.

**Pull request merged**